code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -292,10 +293,19 @@ func (mrh *MinerRestHandler) getNodeStat(w http.ResponseWriter, r *http.Request)
 		common.Respond(w, r, nil, common.NewErrInternal("cannot get latest finalised block"))
 		return
 	}
+	var err error
+	var delegates []event.DelegatePool
+	if includeDelegates {
+		delegates, err = edb.GetDelegatePools(id)
+		if err != nil {
+			common.Respond(w, r, nil, common.NewErrInternal("getting delegates"+err.Error()))
```

Comment:
We should not return internal error If the err is something like gorm.ErrRecordNotFound, which means the provider has no delegates, we can just return empty pools with the node stat to user.